DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfigurationController.cs

```diff
@@ -116,5 +119,100 @@ public IActionResult EditCentreWebsiteDetails(EditCentreWebsiteDetailsViewModel
 
             return RedirectToAction("Index");
         }
+
+        [HttpGet]
+        [Route("RegistrationPrompts/{promptNumber}/Edit")]
+        public IActionResult EditRegistrationPrompt(int promptNumber)
+        {
+            var centreId = User.GetCentreId();
+
+            var customPrompt = customPromptsService.GetCustomPromptsForCentreByCentreId(centreId).CustomPrompts
+                .Single(cp => cp.CustomPromptNumber == promptNumber);
+
+            return View(new EditRegistrationPromptViewModel(customPrompt));
+        }
+
+        [HttpPost]
+        [Route("RegistrationPrompts/{promptNumber}/Edit")]
+        public IActionResult EditRegistrationPrompt(EditRegistrationPromptViewModel model, string action)
+        {
+            if (action.StartsWith("delete") && int.TryParse(action.Remove(0, 6), out var index))
+            {
+                return EditRegistrationPromptPostRemovePrompt(model, index);
+            }
+
+            return action switch
+            {
+                "save" => EditRegistrationPromptPostSave(model),
+                "addPrompt" => EditRegistrationPromptPostAddPrompt(model),
+                _ => View(model)
+            };
+        }
+
+        private IActionResult EditRegistrationPromptPostAddPrompt(EditRegistrationPromptViewModel model)
+        {
+            // We don't want to display validation errors on other fields in this case
+            foreach (var key in ModelState.Keys.Where(k => k != nameof(EditRegistrationPromptViewModel.Answer)))
+            {
+                ModelState[key].Errors.Clear();
+                ModelState[key].ValidationState = ModelValidationState.Valid;
+            }
+
+            if (!ModelState.IsValid)
+            {
+                model.Options = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(model.OptionsString);
+                return View(model);
+            }
+
+            ModelState.Remove(nameof(EditRegistrationPromptViewModel.OptionsString));
+            ModelState.Remove(nameof(EditRegistrationPromptViewModel.Options));
+
+            var (optionsString, options) =
+                NewlineSeparatedStringListHelper.AddStringToNewlineSeparatedList(model.OptionsString, model.Answer!);
+            model.Options = options;
+            model.OptionsString = optionsString;
+
+
+            return View(model);
+        }
+
+        private IActionResult EditRegistrationPromptPostRemovePrompt(EditRegistrationPromptViewModel model, int index)
+        {
+            // We don't want to display validation errors on other fields in this case
+            foreach (var key in ModelState.Keys)
+            {
+                ModelState[key].Errors.Clear();
+                ModelState[key].ValidationState = ModelValidationState.Valid;
+            }

```

Comment:
I wonder whether it would be best to pull this into a method and be explicit -
only clear the errors on the "Add new answer" field and name the method IgnoreAddNewAnswerValidation or something similar.. 🤔

Reply:
Unsure, there are 2 different variants of it now, previously 3, with the different foreach statement